modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -267,9 +277,12 @@ class ResizeSubgraph2 : public ExtractScalesSubgraph
 void simplifySubgraphs(opencv_onnx::GraphProto& net)
 {
     std::vector<Ptr<Subgraph> > subgraphs;
-    subgraphs.push_back(makePtr<UpsampleSubgraph>());
-    subgraphs.push_back(makePtr<ResizeSubgraph1>());
-    subgraphs.push_back(makePtr<ResizeSubgraph2>());
+    subgraphs.push_back(makePtr<UpsampleSubgraph>(true));
+    subgraphs.push_back(makePtr<ResizeSubgraph1>(true));
+    subgraphs.push_back(makePtr<ResizeSubgraph2>(true));
+    subgraphs.push_back(makePtr<UpsampleSubgraph>(false));
+    subgraphs.push_back(makePtr<ResizeSubgraph1>(false));
+    subgraphs.push_back(makePtr<ResizeSubgraph2>(false));
```

Comment:
If we deal Mul-cast according to these changes for Gather-cast then we will have to check 12 subgraphs ? 
And if we deal Mul-cast as in PR #16709 and Gather-cast according to these changes then we would be doing same thing in two different ways.
If you say, I can make changes in PR #16709 similar to this PR, otherwise maybe you can add changes for Mul-cast here in this PR similar to Gather-cast?